packages/rum/src/domain/record/serialization/serializeAttribute.ts

```diff
@@ -30,12 +31,23 @@ export function serializeAttribute(
       case 'placeholder':
         return CENSORED_STRING_MARK
     }
+
     // mask image URLs
-    if (tagName === 'IMG' || tagName === 'SOURCE') {
-      if (attributeName === 'src' || attributeName === 'srcset') {
-        return CENSORED_IMG_MARK
+    if (tagName === 'IMG' && (attributeName === 'src' || attributeName === 'srcset')) {
+      // generate image with similar dimension than the original to have the same rendering behaviour
+      const image = element as HTMLImageElement
+      if (image.naturalWidth > 0) {
+        return censoredImageForSize(image.naturalWidth, image.naturalHeight)
       }
+      const { width, height } = element.getBoundingClientRect()
+      return censoredImageForSize(width, height)
```

Comment:
**💬 suggestion:** ‏if the image is in `display:none` when serialized, the width and height will be 0. If it then become visible, the censored image won't show up because its size will be 0.

Maybe ensure width/height is > 0, or fallback to CENSORED_IMG_MARK